### PR TITLE
esp_lvgl_port: add low power interface. (BSP-371)

### PR DESCRIPTION
**Pull request by espressif2022**
# ESP-BSP Pull Request checklist

> Note: For new BSPs create a PR with this [link](https://github.com/espressif/esp-bsp/compare/main...my-branch?quick_pull=1&template=pr_template_bsp.md).

- [x] Version of modified component bumped
- [x] CI passing

# Change description
add low power interface

### CODE REVIEW

**Comment by espzav** [comment]
@espressif2022 Do you want to help with solving failed tests? I will do review after all tests will pass. 
Note: Main task should be improved by this in future: BSP-266 (there should be more work after LVGL 9 will be released - end of the year)

File: components/esp_lvgl_port/esp_lvgl_port.c [hunk 1]
```diff
@@ -158,6 +158,24 @@ esp_err_t lvgl_port_init(const lvgl_port_cfg_t *cfg)
     return ret;
 }
 
+esp_err_t lvgl_port_resume(void)
+{
+    if (lvgl_port_ctx.tick_timer != NULL) {
+        lv_timer_enable(true);
+        esp_timer_start_periodic(lvgl_port_ctx.tick_timer, lvgl_port_timer_period_ms * 1000);
```
Comment (espzav) [inline components/esp_lvgl_port/esp_lvgl_port.c:165]: Please handle errors and return.

File: components/esp_lvgl_port/esp_lvgl_port.c [hunk 2]
```diff
@@ -158,6 +158,24 @@ esp_err_t lvgl_port_init(const lvgl_port_cfg_t *cfg)
...
+    }
+    return ESP_OK;
+}
+
+esp_err_t lvgl_port_stop(void)
+{
+    if (lvgl_port_ctx.tick_timer != NULL) {
+        lv_timer_enable(false);
+        esp_timer_stop(lvgl_port_ctx.tick_timer);
```
Comment (espzav) [inline components/esp_lvgl_port/esp_lvgl_port.c:174]: Please handle errors and return.

File: components/esp_lvgl_port/esp_lvgl_port.c [hunk 3]
```diff
@@ -535,7 +553,7 @@ static void lvgl_port_task(void *arg)
             task_delay_ms = lv_timer_handler();
             lvgl_port_unlock();
         }
-        if (task_delay_ms > lvgl_port_ctx.task_max_sleep_ms) {
+        if ((task_delay_ms > lvgl_port_ctx.task_max_sleep_ms) || (1 == task_delay_ms)) {
```
Comment (espzav) [inline components/esp_lvgl_port/esp_lvgl_port.c:556]: Why?

**Review by espzav — COMMENTED** [review]
@espressif2022 Thank you for this PR. Some comments left.

File: components/esp_lvgl_port/esp_lvgl_port.c (hunk 3 above)
Comment (espressif2022) [reply components/esp_lvgl_port/esp_lvgl_port.c:556]: > Why?

```
if(lv_timer_run == false) {
    already_running = false; /*Release mutex*/
    return 1;
}
```
after `lv_timer_enable(false)`,` lv_timer_handler()` will return 1 here.

Comment (espzav) [reply components/esp_lvgl_port/esp_lvgl_port.c:556]: OK, thank you for clarification. It should be changed in future. There should be queue and infinity sleep. Wake only when LVGL or touch need it. I have draft for it and still waiting for LVGL 9, which will have callback for wake, when need it.

Comment (espressif2022) [reply components/esp_lvgl_port/esp_lvgl_port.c:558]: > OK, thank you for clarification. It should be changed in future. There should be queue and infinity sleep. Wake only when LVGL or touch need it. I have draft for it and still waiting for LVGL 9, which will have callback for wake, when need it.

Okay, currently lvgl 8.3, I don’t see any other low-power input control interface now.
 Looking forward to version 9.0.

File: components/esp_lvgl_port/esp_lvgl_port.c [hunk 4]
```diff
@@ -158,6 +158,26 @@ esp_err_t lvgl_port_init(const lvgl_port_cfg_t *cfg)
...
 }
 
+esp_err_t lvgl_port_resume(void)
+{
+    if (lvgl_port_ctx.tick_timer != NULL) {
+        lv_timer_enable(true);
+        return esp_timer_start_periodic(lvgl_port_ctx.tick_timer, lvgl_port_timer_period_ms * 1000);
+    } else {
+        return ESP_ERR_INVALID_ARG;
+    }
```
Comment (espzav) [inline components/esp_lvgl_port/esp_lvgl_port.c:168]: I am sorry, but it is not nice. Better can be second `return` without `else` OR put into `err = esp_timer_star_...` and then `return err` on the end of function;

File: components/esp_lvgl_port/esp_lvgl_port.c [hunk 5]
```diff
@@ -158,6 +158,26 @@ esp_err_t lvgl_port_init(const lvgl_port_cfg_t *cfg)
...
+}
+
+esp_err_t lvgl_port_stop(void)
+{
+    if (lvgl_port_ctx.tick_timer != NULL) {
+        lv_timer_enable(false);
+        return esp_timer_stop(lvgl_port_ctx.tick_timer);
+    } else {
+        return ESP_ERR_INVALID_ARG;
+    }
```
Comment (espzav) [inline components/esp_lvgl_port/esp_lvgl_port.c:178]: Same as in previous function...

**Review by espzav — APPROVED** [review]
Great! Thank you very much! LGTM

**Pull request merged**